### ISSUE TITLE
KTF Resource Cleanup

### ISSUE THREAD

**Issue by shaneutt** [Kong/kubernetes-ingress-controller#2618]
### Problem Statement

Historically in tests we've written multiple defer functions which come in at several lines of code for each resource we want to clean up from a test run to ensure that any particular test run doesn't leave behind orphaned resources upon success or failure. With recent releases of [KTF](https://github.com/kong/kubernetes-testing-framework) a `clusters.Cleaner` type was added that can be used to dynamically and automatically clean up resources created by a test and the purpose of this task is to refactor our tests onto this tool to significant reduce test sprawl.

**Comment by Raving-hash** [comment]
I find KTF cleaner is used in the most of the setup and cleanup,so this issue maybe close now?

**Comment by pmalek** [comment]
That's correct @Raving-hash. All of the usages of `defer` in tests is for not related to resource cleanup, [which is done using the cleaner](https://github.com/Kong/kubernetes-ingress-controller/blob/3a22b4a1ba5dfa3c81261c002701ec5077314270/test/integration/suite_test.go#L117-L122). 

Thanks for noticing 🙇
